### ISSUE TITLE
Add a native spreadsheet

### ISSUE THREAD

**Issue by sandraev** [UpsilonNumworks/Upsilon#87]
Using mainly KhiCAS for its spreadsheet, I would like to be able to use it without having to install the whole application (whose periodic table is not really useful anymore...), and with a more native design.

So I was thinking that creating a spreadsheet application could be a solution, either external, or internal, while joining issue #75...

**Comment by Overengined** [comment]
I agree, it would be cool to have a spreadsheet app with some basic formulas and reactivity.

**Comment by Lauryy06** [comment]
It's indeed a really nice feature, but also very long and hard to implement. So I will not work on it for the moment, but I keep the idea in my mind !

**Comment by Overengined** [comment]
> It's indeed a really nice feature, but also very long and hard to implement. So I will not work on it for the moment, but I keep the idea in my mind !

I suggest you add it on the roadmap 😉

**Comment by sandraev** [comment]
[sandraev/upsilon-py-spreadsheet](https://github.com/sandraev/upsilon-py-spreadsheet)

As I don't have (yet!) a good command of C++, I made a little spreadsheet model in Python.
Of course, it's nothing like a real application, but if it can give a structure to build on...
Granted, it's not beautiful OOP, organized with MVC and all, but Python is a bit limited in this area.
The GUI is hand-made, there's no scrolling (Escher will be soon in Python? ;-) and I had the laziness to reprogram Poincare for the calculations, but otherwise, it looks like it.
I hope this will help you, until I get serious about it.

**Comment by Overengined** [comment]
It looks quite cool

**Comment by sandraev** [comment]
Thank you! But I hope it will be useful!
